File: Data-Structures/Arrays/Exercises/Valid-Anagram/Solution-2/solution2.cpp
#include <iostream>
#include <set>
#include <algorithm>
#include <string>

using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length())
            return false;
            
        sort(s.begin(),s.end());
        sort(t.begin(),t.end());

        for (int i=0; i < t.length(); i++)
            if (s[i] != t[i])
                return false;

        return true;
    }
};

int main() {
    cout << Solution().isAnagram("anagram", "nagaram") << "\n";
    cout << Solution().isAnagram("rat", "car") << "\n";
    return 0;
}

File: Data-Structures/Arrays/Exercises/Valid-Anagram/Solution-2-Sort/solution2.py
import unittest

###### IMPLEMENTATION #######

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        return ''.join(sorted(s)) == ''.join(sorted(t))

###### TESTING #######

class TestIsAnagram(unittest.TestCase):
    def test_1(self):
        self.assertEqual(Solution().isAnagram("anagram", "nagaram"), True)
    
    def test_2(self):
        self.assertEqual(Solution().isAnagram("rat", "car"), False)


if __name__ == '__main__':
    unittest.main()